homeassistant/components/statistics/sensor.py

```diff
@@ -393,11 +388,22 @@ def _derive_unit_of_measurement(self, new_state: State) -> str | None:
     @property
     def device_class(self) -> SensorDeviceClass | None:
         """Return the class of this device."""
-        if self._state_characteristic in STAT_NUMERIC_RETAIN_UNIT:
-            _state = self.hass.states.get(self._source_entity_id)
-            return None if _state is None else _state.attributes.get(ATTR_DEVICE_CLASS)
         if self._state_characteristic in STATS_DATETIME:
             return SensorDeviceClass.TIMESTAMP
+        if self._state_characteristic in STATS_NUMERIC_RETAIN_UNIT:
+            _source_state = self.hass.states.get(self._source_entity_id)
+            if _source_state is None:
+                return None
+            source_device_class = _source_state.attributes.get(ATTR_DEVICE_CLASS)
+            if source_device_class is None:
+                return None
+            sensor_device_class = SensorDeviceClass(source_device_class)
```

Comment:
This may raise. Use `homeassistant.util.enum.try_parse_enum`.

Reply:
thanks for the tip!